src/core/loss_detection.c

```diff
@@ -1450,6 +1531,13 @@ QuicLossDetectionProcessAckBlocks(
 
     LossDetection->ProbeCount = 0;
 
+    AckedPacketsIterator = AckedPackets;
+    while (AckedPacketsIterator != NULL) {
+        QUIC_SENT_PACKET_METADATA* Packet = AckedPacketsIterator;
+        AckedPacketsIterator = AckedPacketsIterator->Next;
+        QuicSentPacketPoolReturnPacketMetadata(&Connection->Worker->SentPacketPool, Packet);
+    }
```

Comment:
Why do we need to add a new clean up path?

Reply:
Because we use the acked packet info list for `QuicCongestionControlOnDataAcknowledged` function of BBR.

We have to delay the time for the clean up work until it's get done.